src/gui/search/searchsortmodel.cpp

```diff
@@ -29,17 +29,89 @@
 #include "searchsortmodel.h"
 
 SearchSortModel::SearchSortModel(QObject *parent)
-    : QSortFilterProxyModel(parent)
+    : base(parent)
+    , m_isNameFilterEnabled(false)
+    , m_minSeeds(0)
+    , m_maxSeeds(-1)
+    , m_minLeechs(0)
+    , m_maxLeechs(-1)
+    , m_minSize(0)
+    , m_maxSize(-1)
 {
 }
 
+void SearchSortModel::enableNameFilter(bool enable)
+{
+    m_isNameFilterEnabled = enable;
+}
+
+void SearchSortModel::setNameFilter(const QString &searchTerm)
+{
+    m_searchTerm = searchTerm;
+    if (searchTerm.length() > 2 &&
+            searchTerm.at(0) == QChar('\"') && searchTerm.at(searchTerm.length() - 1) == QChar('\"')) {
+        m_searchTermWords.append(m_searchTerm.mid(1, m_searchTerm.length() - 2));
+    }
+    else {
+        m_searchTermWords = searchTerm.split(QChar(' '), QString::SkipEmptyParts);
+    }
+}
+
+void SearchSortModel::setSizeFilter(qint64 minSize, qint64 maxSize)
+{
+    m_minSize = std::max(static_cast<qint64>(0), minSize);
```

Comment:
My compiler (GCC 5.3) does not agree with you. I shall either write:
`std::max(0ll, minSize)`, which does not seem nice to me, since `long long` could be not `int64`, or something like
`std::max(quint64{0}, minSize)`, which does not differ much from `static_cast`.